src/XeroPHP/Remote/Query.php

```diff
@@ -135,22 +221,45 @@ public function execute() {
             $request->setParameter('offset', $this->offset);
         }
 
-        $request->send();
-
-        $elements = array();
-        foreach($request->getResponse()->getElements() as $element) {
-            /** @var Object $built_element */
-            $built_element = new $from_class($this->app);
-            $built_element->fromStringArray($element);
-            $elements[] = $built_element;
+        if($this->includeArchived) {
+            $request->setParameter('includeArchived', $this->includeArchived);
         }
 
-        return $elements;
+        return $request;
     }
 
     /**
-     * @return mixed
+     * @return string
      */
+    private function getParsedWhere()
+    {
+        return preg_replace_callback('/:[_a-zA-Z0-9]+/', function($matches) {
+            $parameterKey = ltrim($matches[0], ':');
+            $parameterValue = $this->parameters[$parameterKey];
+
+            return $this->convertPHPValueToXeroQuery($parameterValue);
+        }, $this->getWhere());
+    }
+
+    /**
+     * @param string $value
+     * @return string
+     */
+    private function convertPHPValueToXeroQuery($value)
+    {
+        if ($value instanceof \DateTime) {
+            return sprintf('DateTime(%s)', $value->format('Y,m,d'));
+        } else if (is_int($value) || is_float($value)) {
+            return $value;
```

Comment:
In this line, you're basically saying if it's an int or a float, return it.  Even though the function returns a 'string'  I think it'd be appropriate to cast it here.  It's a shame we can't share almost duplicate functionality [here](https://github.com/calcinai/xero-php/blob/master/src/XeroPHP/Remote/Object.php#L227)